python/lsst/meas/algorithms/brightStarStamps.py

```diff
@@ -0,0 +1,280 @@
+    >>> starSubregions = butler.get("brightStarStamps_sub", dataId, bbox=bbox)
+    """
+
+    def __init__(self, starStamps, innerRadius=None, outerRadius=None,
+                 metadata=None,):
+        self._starStamps = [self._checkStarStamp(stamp) for stamp in starStamps]
+        self._setMetadata(metadata)
+        # Add inner and outer radii to metadata
+        self._setAnnulusRadii(innerRadius, outerRadius)
+
+    def __len__(self):
+        return len(self._starStamps)
+
+    def __getitem__(self, index):
+        return self._starStamps[index]
+
+    def append(self, item):
+        """Add an additional bright star stamp.
+
+        Parameters
+        ----------
+        item : `dict`
+            Mapping containing keys ``"starStamp"``, ``"gaiaGMag"``,
+            ``"gaiaId"``, and `"annularFlux"`.
+        """
+        self._starStamps.append(self._checkStarStamp(item))
+        return None
+
+    def getMaskedImages(self):
+        """Retrieve star images.
+
+        Returns
+        -------
+        maskedImages :
+            `list` [`lsst.afw.image.maskedImage.maskedImage.MaskedImageF`]
+        """
+        return [stamp["starStamp"] for stamp in self._starStamps]
+
+    def getMagnitudes(self):
+        """Retrieve Gaia G magnitudes for each star.
+
+        Returns
+        -------
+        gaiaGMags : `list` [`float`[
+        """
+        return [stamp["gaiaGMag"] for stamp in self._starStamps]
+
+    def getGaiaIds(self):
+        """Retrieve Gaia IDs for each star.
+
+        Returns
+        -------
+        gaiaIds : `list` [`str`]
+        """
+        return [str(stamp["gaiaId"]) for stamp in self._starStamps]
+
+    def getAnnularFluxes(self):
+        """Retrieve normalization factors for each star.
+
+        These are computed by integrating the flux in annulus centered on the
+        bright star, far enough from center to be beyond most severe ghosts and
+        saturation. The inner and outer radii that define the annulus can be
+        recovered from the metadata.
+
+        Returns
+        -------
+        annularFluxes : list[`float`]
+        """
+        return [stamp["annularFlux"] for stamp in self._starStamps]
+
+    def selectByMag(self, magMin=None, magMax=None):
+        """Return the subset of bright star stamps for objects with specified
+        magnitude cuts (in Gaia G).
+
+        Parameters
+        ----------
+        magMin : `float`, optional
+            Keep only stars fainter than this value.
+        magMax : `float`, optional
+            Keep only stars brighter than this value.
+        """
+        selector = np.array([True for _ in range(len(self))])
+        if magMin is not None:
+            selector *= np.array(self.getMagnitudes()) > magMin
+        if magMax is not None:
+            selector *= np.array(self.getMagnitudes()) < magMax
+        selIdx = np.where(selector)[0]
+        return BrightStarStamps([self._starStamps[j] for j in selIdx], metadata=self._metadata)
```

Comment:
This is great. I learned so much from this - thanks Nate!

I went with _all_ of your suggestions, including folding in the retrieval of magnitudes to avoid the GetMagnitudes loop. I think it still makes for readable code - at least much more so than my initial 13 loops.